rocAL/source/pipeline/master_graph.cpp

```diff
@@ -1804,3 +1826,147 @@ void MasterGraph::serialize(size_t *serialized_string_size) {
     _pipeline_serializer.serialize_to_string(_serialized_pipeline);
     *serialized_string_size = _serialized_pipeline.size();
 }
+
+uint64_t MasterGraph::compute_pipeline_signature() const {
```

Comment:
[Non-blocking] `compute_pipeline_signature()` silently skips any `Argument` values whose `typeid` is not in the handled set, since they fall through to the `catch (...)`. That works today, but it makes the signature dependent on which value types happen to be supported and can hide future mismatches. It might be worth at least logging or asserting on unexpected types in debug builds so new argument types dont accidentally stop contributing to the signature.